waiter/src/waiter/scheduler.clj

```diff
@@ -481,10 +481,11 @@
                                         ; Assume nil service-instance-info means there was a failure in invoking marathon
                                         (conj scheduler-messages
                                               [:update-service-instances
-                                               (assoc service-instance-info
-                                                 :service-id service-id
-                                                 :failed-instances all-failed-instances
-                                                 :scheduler-sync-time request-instances-time)])
+                                               (merge service-instance-info
+                                                      (select-keys service [:instances :task-count])
```

Comment:
Can we destructure on `service` above and continue to use `assoc`?

Reply:
Done.